pkg/planner/core/find_best_task.go

```diff
@@ -264,8 +264,16 @@ func iteratePhysicalPlan4BaseLogical(
 	childTasks = childTasks[:0]
 	// The curCntPlan records the number of possible plans for selfPhysicalPlan
 	curCntPlan := int64(1)
+	var isParentPhyscicalHashJoin bool
```

Comment:
The better way to add strict in this issue case is that "Check the current operator is HashAgg instead of check the child operator is HashJoin"

Reply:
I have been updated.